src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -66,6 +66,11 @@ public void Add(string hintName, SourceText source)
                 throw new ArgumentException(CodeAnalysisResources.HintNameUniquePerGenerator, nameof(hintName));
             }
 
+            if (source.Encoding is null)
```

Comment:
In the case this is a `StringSourceText` though we could just add `Utf8` ourselves and avoid the exception. Can do this later if that is a strong source of feedback